server/Models/getRestaurantInfo.js

```diff
@@ -0,0 +1,68 @@
+const pg = require('../db');
+
+const getRestaurantInfo = (req, res) => {
+  const { restaurantId } = req.params;
+
+  const { sort, keyword, star } = req.body;
+  let orderByStr = '';
+  let selectStr = orderByStr;
+  let sortDirection = '';
+  let keywordStr = '';
+  let starStr = '';
+
+  if (sort === 'highest rating') {
+    orderByStr = 'CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4';
+    sortDirection = ' DESC,';
+  }
+
+  if (sort === 'lowest rating') {
+    orderByStr = 'CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4';
+    sortDirection = ' ASC,';
+  }
+
+  if (keyword !== undefined) {
+    for (let i = 0; i < keyword.length; i ++) {
+      keywordStr += `AND reviews.reviewText LIKE '%${keyword[i]}%' `;
+    }
+
+    keywordStr = keywordStr.slice(0, -1);
+  }
+
+  if (orderByStr !== '') {
+    selectStr = ', ' + orderByStr + ' AS avg_overal_from_reviews';
```

Comment:
if you have the alias in the select statement, you don't have to recompute the avg value in the where statement

Reply:
see the comment below.